src/devices/RGBLedMatrix/RgbLedMatrix.cs

```diff
@@ -0,0 +1,641 @@
+        }
+
+        /// <summary>
+        /// Fill the whole display area with a specific color
+        /// </summary>
+        /// <param name="red">red color value</param>
+        /// <param name="green">green color value</param>
+        /// <param name="blue">blue color value</param>
+        /// <param name="backBuffer">true if to draw on back buffer, false to draw on the forground buffer</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe void Fill(byte red, byte green, byte blue, bool backBuffer = false)
+        {
+            FillRectangle(0, 0, Width, Height, red, green, blue, backBuffer);
+        }
+
+        /// <summary>
+        /// Set color of specific pixel on the forground buffer display
+        /// </summary>
+        /// <param name="column">x coordinate of the pixel</param>
+        /// <param name="row">y coordinate of the pixel</param>
+        /// <param name="red">red color value</param>
+        /// <param name="green">green color value</param>
+        /// <param name="blue">blue color value</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void SetPixel(int column, int row, byte red, byte green, byte blue)
+        {
+            SetPixel(column, row, red, green, blue, _colorsBuffer);
+        }
+
+
+        /// <summary>
+        /// Set color of specific pixel on the background buffer display
+        /// </summary>
+        /// <param name="column">x coordinate of the pixel</param>
+        /// <param name="row">y coordinate of the pixel</param>
+        /// <param name="red">red color value</param>
+        /// <param name="green">green color value</param>
+        /// <param name="blue">blue color value</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void SetBackBufferPixel(int column, int row, byte red, byte green, byte blue)
+        {
+            SetPixel(column, row, red, green, blue, _colorsBackBuffer);
+        }
+
+        private void SetPixel(int column, int row, byte red, byte green, byte blue, byte [] colorsBuffer)
+        {
+            if ((column | row) < 0 || column >= Width || row >= Height)
+            {
+                return;
+            }
+
+            red = s_gamma[red];
+            green = s_gamma[green];
+            blue = s_gamma[blue];
+
+            if (_chainRows > 1)
+            {
+                int panelRow = row / _deviceRows;
+
+                if ((_chainRows & 1U) == 1 ^ (panelRow & 1U) == 0)
+                {
+                    int panelWidth = Width / _chainColumns;
+                    int panelColumn = column / panelWidth;
+
+                    column = column % panelWidth + (_chainColumns - 1 - panelColumn) * panelWidth;
+                }
+
+                row = row % _deviceRows;
+                column = panelRow * Width + column;
+            }
+
+            int pos = 8 * column + 8 * (row % (_deviceRows >> 1)) * _fullChainWidth;
```

Comment:
is it same formula for 32x32 and 64x64?

Reply:
yes this work different sizes. The reason is we always render 2 lines in the same time. ```(_deviceRows >> 1)``` always will make it work correctly.